### ISSUE TITLE
Types for initial state broken in master

### ISSUE THREAD

**Issue by p-netm** [onaio/reveal-frontend#781]
build is failing for quite a few store/ducks module for the type annotations of the initialstate.

**Comment by moshthepitt** [comment]
what caused this? could you share a stacktrace?

**Comment by cKellyDesign** [comment]
@p-netm please update this ticket with the results of your sleuthing to shed some light as to why tests are breaking on master. 🙏

**Comment by p-netm** [comment]
Sample stacktrace from `goals`.

```
Type 'ImmutableObject<{ currentGoal: null; goalsById: {}; }>' is not assignable to type 'ImmutableGoalState'.
  Type 'ImmutableObject<{ currentGoal: null; goalsById: {}; }>' is not assignable to type 'GoalState'.
    Types of property 'goalsById' are incompatible.
      Type 'ImmutableObject<{}>' is not assignable to type '{ [key: string]: Goal[]; }'.
        Index signature is missing in type 'ImmutableObjectMixin<{}> & {}'.ts(2322)
```

**Comment by moshthepitt** [comment]
@p-netm Thanks.  What change led to this to start happening?

**Comment by p-netm** [comment]
At the moment I don't know, I can time travel through the commits, but I fear it might be a lot of work.

**Comment by moshthepitt** [comment]
> At the moment I don't know, I can time travel through the commits, but I fear it might be a lot of work.

Yeah don't do that

**Comment by cKellyDesign** [comment]
@p-netm do we have a clear fix?

**Comment by p-netm** [comment]
> @p-netm do we have a clear fix?

@cKellyDesign This should fix it f0acb2f

**Comment by p-netm** [comment]
Delete yarn.lock, run yarn .
this opens another can of worms typewise
